src/transform.ts

```diff
@@ -69,6 +71,20 @@ export class Transform {
         continue;
       }
 
+      if (options?.peerProject) {
+        const contracts = [...findAll('ContractDefinition', output.sources[source].ast)];
+
+        for (const noContract of contracts.filter(
+          ({ contractKind }) => contractKind !== 'contract',
+        )) {
+          this.getData(noContract).importPath = path.join(options.peerProject, source);
+        }
+
+        if (contracts.every(({ contractKind }) => contractKind !== 'contract')) {
+          continue;
+        }
+      }
```

Comment:
My main thought after a quick look through the PR is that I don't think this logic belongs here. This new code is adding highly domain-specific transpilation concerns into `Transform`, but I think the `Transform` class should remain generic and agnostic of the actual transpilation logic.

I don't think the `Transform` class needs to change for this feature. We can use the existing exclusion mechanism. You should look at the way it works for `findAlreadyInitializable`:

https://github.com/OpenZeppelin/openzeppelin-transpiler/blob/4238e3690ffa29004e671a75ddd503616c543c22/src/index.ts#L71-L80

We can define a similar helper to find the files that should be excluded. We don't really need to use the `getData` mechanism here. The helper would run prior to the transformation and return a Map, mapping ContractDefinition nodes to the `importPath`/`useFromPartial` value, then we pass this as a parameter to the transformations that need it.

Reply:
There are two things here:
- identifying which nodes should not be transpiled, and set the `importFromPeer` value in `getData`
- files for which no node should be transpiled should be excluded.

I guess it would be possible to construct the `exclude` function differently in `transpile` as you proposed ... but what about the initial population of `getData` ?

My initial ideal was: 
- In transpile, loop over the solcOutput to analyse the contract declaration
- populate an `peerExcluded = string[]` array that could be used in the exclude lambda
- call `getData` after the `Transform` is created, but before the transforms are applied to set the data. That can't be done using a transform because the transforms only have the source unit. We need the ast of everything (including excluded contract) to do that.

Reply:
Let me know what you think of the latest commit